### DISCUSSION TITLE
Retetnion

### DISCUSSION THREAD

**Discussion by sreejesh-radhakrishnan-db** [strimzi/strimzi-kafka-operator#7422] [Q&A] [answered]
Hi

Can I make some topic be retained for ever? Until  I drop and recreated the topic?   we have a unique situation for the  topic to hold moment of truth data, which till recreated need to be retained on topic?  is that something possible?

**Comment by ppatierno** [comment, marked as answer]
From a configuration this is something possible with Kafka.
You should set the `log.retention.ms` at `-1` which will apply no time limit. More info [log.retention.ms](https://kafka.apache.org/documentation/#brokerconfigs_log.retention.ms)
Taking into account that the `log.retention.bytes` is `-1` by default, you should have retention "forever" and not deleting on size or time.
Of course, it could be dangerous because your disk could be run out of space if log segments are not deleted periodically and you could have a not working broker soon (even not able to start).

**Reply by sreejesh-radhakrishnan-db** [reply]
thank you

**Reply by sreejesh-radhakrishnan-db** [reply]
log.retention.hours: 336  is set on the CLUSTER/BROKER level , and on topic level I am superseding where I want forever log retention  by  log.retention.ms=-1 , that will work right.  just cross checking
